realm/realm-library/src/androidTest/java/io/realm/internal/CollectionTests.java

```diff
@@ -396,15 +398,16 @@ public void onChange(Collection collection1) {
                         assertEquals(collection1.size(), 4);
                         break;
                     case 1:
+                        assertFalse(transactionCommitted.get());
                         assertEquals(collection1.size(), 5);
                         sharedRealm.close();
-                        looperThread.testComplete();
                         break;
                 }
             }
         });
         addRow(sharedRealm);
-        assertEquals(collection.size(), 5);
+        transactionCommitted.set(true);
+        looperThread.testComplete();
```

Comment:
I think we should check if that the listener is called twice.
I don't think this updated test fails in case of not being called when committed.